classes/class-wc-rest-connect-address-normalization-controller.php

```diff
@@ -96,11 +105,29 @@ public function update_items( $request ) {
 		$response->normalized->name = $name;
 		$response->normalized->company = $company;
 		$response->normalized->phone = $phone;
+		$is_trivial_normalization = isset( $response->is_trivial_normalization ) ? $response->is_trivial_normalization : false;
+
+		if ( $is_trivial_normalization ) {
+			$this->update_address( $order_id, $type, $response->normalized );
+		}
 
 		return array(
 			'success' => true,
 			'normalized' => $response->normalized,
-			'is_trivial_normalization' => isset( $response->is_trivial_normalization ) ? $response->is_trivial_normalization : false,
+			'is_trivial_normalization' => $is_trivial_normalization,
+		);
+	}
+
+	public function accept_suggestion( $request ) {
+		$data     = $request->get_json_params();
+		$order_id = $data[ 'orderId' ];
+		$type     = $data[ 'type' ];
+		$address  = $data[ 'address' ];
+
+		$this->update_address( $order_id, $type, $address );
```

Comment:
What about doing this when the actual purchase is made, or in the `Rates` request? 1 less endpoint, 1 less round-trip to the server, less complexity.